### ISSUE TITLE
Crashed when training the referenced imagenet model, and cannot reload from the snapshot

### ISSUE THREAD

**Issue by byzhang** [BVLC/caffe#1086]
Hi,
The training crashed with this output:
I0915 20:04:13.175724 23440 solver.cpp:196] Iteration 123000, loss = 1.63368
I0915 20:04:13.175762 23440 solver.cpp:211]     Train net output #0: loss = 1.63368 (\* 1 = 1.63368 loss)
I0915 20:04:13.175775 23440 solver.cpp:407] Iteration 123000, lr = 0.001
[1]    23440 segmentation fault (core dumped)  ./build/tools/caffe train  --snapshot

And then I tried to reload it:
`./build/tools/caffe train --solver=models/bvlc_reference_caffenet/solver.prototxt --snapshot models/bvlc_reference_caffenet/caffenet_train_iter_120000.caffemodel`
It failed with this output:
I0915 23:45:27.799008  2939 caffe.cpp:105] Resuming from models/bvlc_reference_caffenet/caffenet_train_iter_120000.caffemodel
I0915 23:45:27.799017  2939 solver.cpp:165] Solving CaffeNet
I0915 23:45:27.799041  2939 solver.cpp:170] Restoring previous solver status from models/bvlc_reference_caffenet/caffenet_train_iter_120000.caffemodel
F0915 23:45:27.991425  2939 io.cpp:51] Check failed: fd != -1 (-1 vs. -1) File not found: fc8labelloss"loss(�
**\* Check failure stack trace: ***
    @     0x7f1abfda1daa  (unknown)
    @     0x7f1abfda1ce4  (unknown)
    @     0x7f1abfda16e6  (unknown)
    @     0x7f1abfda4687  (unknown)
    @           0x496ab3  caffe::ReadProtoFromBinaryFile()
    @           0x45a87d  caffe::Solver<>::Restore()
    @           0x45a9d4  caffe::Solver<>::Solve()
    @           0x4161ef  train()
    @           0x410a21  main
    @     0x7f1abbb6bec5  (unknown)
    @           0x414a57  (unknown)
    @              (nil)  (unknown)
[1]    2939 abort (core dumped)  ./build/tools/caffe train  --snapshot

The branch is
- dev    c69b3b4 Merge pull request #1051 from jeffdonahue/travis-red-errors

Any suggestions on how to resume the training, and what'd be the cause of the first crash?
